platform/engineering/design-docs/2020-08-11-redirects-from-cms.md

```diff
@@ -0,0 +1,224 @@
+
+
+### Testing Plan
+The current content release method already does link checking.  This method would be tested to a large degree by this existing test.  The CMS already does a significant amount of testing of a front end build in a non-AWS environment so we would add a Behat test to makes sure the that the client side redirect is working.
+
+Additional testing could be added on the Vets-website deployment tests to verify that a set of redirects work properly on the server side redirects in AWS.
+
+### Logging
+Logging would not be required.  There may be some statistics that could be tracked within AWS, but that may not be necessary.  Google Analytics may also be used to track instances of displaying the 404 page that may in some cases include referral data.  Server logs track HTTP requests and are likely already used to track requests ending in 404 errors.
+
+### Debugging
+CURL would be the simplest and most useful debugging method of looking at and debugging redirects on any of the BRD environments as well as local environments.  On local environments simply inspecting a page's meta refresh tag would provide debug specific data.
+
+### Caveats
+Section level redirects that are outside of the CMS could still be handled as they are at the server or F5 level.
+
+### Security Concerns
+None at this time.
+
+### Privacy Concerns
+None at this time.
+
+### Open Questions and Risks
+
+1. **Do we still need `Type` and `Match Type` redirects in the CMS?** Back when the notion of redirects being imported into the CMS was first planned for, the CMS team was asked to add `Type` (Server | Template) and `Match Type` (prefix | regex-i | regex | non re ) data to redirect entries.  That was added by creating the [redirect options](https://www.drupal.org/project/redirect_options) module.  I am not certain at this time if this data would still be needed for this approach or not.  It seemed to be for managing redirects beyond the content scope of the CMS.  The CMS can support that but it is not required and also not blocked by this proposal. [Related to this dropped epic](https://github.com/department-of-veterans-affairs/va.gov-team/issues/503).
+
+### Work Estimates
+1. CMS: Add destination path to CMS content export. (~3 story points)
+2. PWS: Add build of HTML pages that have meta-refresh along side the current HTML page builds for each content export redirect.*.json file.  These create the client side redirects for use in non-AWS environments. (~3 story points)
+3. PWS Devops : Add processing to vets-website deploy to create an AWS Object with redirect metadata for any HTML page that contains a meta-refresh tag. This would create server side redirects for any AWS environments. (~10 story points)
+
+### Alternatives
+
+#### Import redirects from CMS into nginX or F5
+This [epic was planned and then dropped](https://github.com/department-of-veterans-affairs/va.gov-team/issues/503).
```

Comment:
I heard a rumor that this approach was shot down because having Drupal as central authority doesn't allow a place to inject any review of URLs being used for redirects --- that any content editor can create URL redirects without oversight. is this true? if so, how does this approach address that?

Reply:
There are templates/content types that oversight is required (i.e. core benefit content, VAMC location pages), there are other cases where we may let them happen organically (i.e. new articles)

Reply:
I did not see that rumor listed on the epic that was closed.  Drupal can have all kinds of workflows.  If there is a moderation of redirects that is needed, it could be added.  However, there is limited oversight now if an editor changes the name on a page that page gets moved to a different path. For some editors and some pieces of content (depending on its owner) the change of title goes through editorial review, while others can just make the change and publish it.  Either way,  they have likely created a 404.  

Other than the automated creation of redirects stated above, the other CRUD operations  of redirects are permed and currently assigned  to roles in the CMS 
![image](https://user-images.githubusercontent.com/5752113/91337560-89733080-e7a1-11ea-9227-d87abb516532.png)

Reply:
Please double check this, as far as I am aware, an H1 change does not always result in an automatic URL change in certain content templates - namely benefit content.

Reply:
Right TItles  in Drupal (H1 are up to the front end) are not tied to paths on all content types. They are on several.